client-v2/src/shared/components/input/CreateInput.tsx

```diff
@@ -44,20 +52,59 @@ export default (
     React.HTMLAttributes<HTMLInputElement> & StyledProps<any>
   >,
   type?: string
-) => ({ label, input, originalValue, ...rest }: Props) => (
-  <InputContainer>
-    {label && (
-      <InputLabel htmlFor={label}>
-        {label}
-        {originalValue && input.value !== originalValue && (
-          <RewindButton onClick={() => input.onChange(originalValue)}>
-            <RewindIcon />
-          </RewindButton>
-        )}
-      </InputLabel>
-    )}
-    <InputComponentContainer>
-      <Component id={label} {...input} {...rest} type={type} />
-    </InputComponentContainer>
-  </InputContainer>
-);
+) => {
+  return class extends React.Component<Props, State> {
+    private inputElement: React.RefObject<HTMLInputElement>;
+    constructor(props: Props) {
+      super(props);
+      this.inputElement = React.createRef();
+      this.state = { inputHeight: 36 };
```

Comment:
Can we always rely on this value staying in sync with our styles? I wonder whether we can derive it from constants/theme (if there isn't a value for inputs, we could create one), but I'm also wondering whether it's necessary at all --- is there some strange pop-in behaviour when the component first renders, or similar, if it isn't there? We could always avoid adding the height in the `style` prop below if we don't have a value in the state yet.

Reply:
Ah yup - will look up the themes to check if there is a pre-defined value.
I think it is beneficial to keep a default there, but will investigate.

Reply:
The textarea element provides its own default height of 2 rows. We can use this by allowing the`inputHeight` to default to the 'auto' value, rather than specifying a number ourselves.